### DISCUSSION TITLE
Multi-cloud

### DISCUSSION THREAD

**Discussion by jarrodmoldrich** [ex-aws/ex_aws#926] [Q&A] [answered]
Hello ex_aws people,

Thank you for maintaining this library.

I'd like to use ex_aws for both AWS and AWS compatible services, like B2.  Is it possible to have separate credentials for different services?

Cheers,

Jarrod

**Comment by bernardd** [comment, marked as answer]
You can already pass in custom config values to any `ExAws.request` call. Did you have something else in mind?

**Comment by jarrodmoldrich** [comment]
Hi @bernardd ,

Thank you for the answer.  I found this also by looking a bit closer at the source code, but I haven't had the time to try it out.  Normally I try to figure out things on my own if I can before I ask a question, but It didn't seem obvious how to do it just by looking at the docs, so I posted here.

It also seems like the authentication config can be modified per-service, like this:

```
config :ex_aws, :s3,
  access_key_id: "<key_id>",
  secret_access_key: "<access_key>"
```

But I haven't had the time to try this out yet.

Jarrod
